Add support for numeric and timestamp to RJDBC and include joda library dependency

perNyfelt opened the pull request:
Reference issue: #447 and #446

Added fix for numeric and timestamp and the missing joda dependency. Also upgraded some versions. I tried with DBI 1.0.0-b9 and that did work for RJDBC but I do not know the details of mysql setup to test so could not verify it so reverted back to what is was before (0.7-b4) to not risk breaking anything.

perNyfelt committed:
Enable RJDBC to handle TIMESTAMP and NUMERIC

perNyfelt committed:
This works fine one Linux so enabling it

perNyfelt committed:
Remove renjin-hamcrest-maven-plugin so I can create a proper pull request

perNyfelt committed:
Add info on RJDBC use

akbertram commented on RJDBC/pom.xml:
```diff
@@ -19,12 +19,24 @@
+			<version>2.10.1</version>
+		</dependency>
+		<dependency>
+			<groupId>com.h2database</groupId>
```
Why add a dependency on h2database to RJDBC?
We should have a separate module for h2.

perNyfelt replied:
This was to verify that the addition of TIMESTAMP worked. It was just a test dependency. To avoid confusion on "why is this not in h2 module" I changed the test to use Apache Derby instead. Hope that makes it clearer.

akbertram commented on RJDBC/src/test/R/TestH2.R:
```diff
@@ -0,0 +1,43 @@
+library("hamcrest")
+
+
+drv <- JDBC("org.h2.Driver")
```
Move to h2 module

perNyfelt replied:
Changed to Derby. See comment above.

perNyfelt committed:
Changed test to Derby

perNyfelt commented:
I actually had problems with DBI 0.7-b4 so suggest an upgrade to 1.0.0-b9. The issue was

Evaluating '.../renjin-dbi/RJDBC/src/main/R/JDBCConnection.R'
Error : the 'subClass' list for class "DBIConnection", includes an undefined class "AnsiConnection"
ERROR: error in contained classes ("DBIConnection") for class "JDBCConnection"; class definition removed from 'RJDBC'. 

This problem goes away when using 1.0.0-b9.

perNyfelt committed:
ignore DerbyDB dirs

perNyfelt committed:
Additional fixes to datatypes in RJDBC
Switched test plugin to be able to start fixing the other issues with
the other DBI modules.
Bumped up version to match the latest official release +1 -SNAPSHOT

perNyfelt committed:
Add note on Windows build issues

perNyfelt committed:
set joda dependency to provided

perNyfelt committed:
Make DPI provided

perNyfelt committed:
Add DBI dependency to plugins

perNyfelt committed:
Fix wrong handling of datetime / timestamps (java.sql.Date truncates the hour, minute, sec etc. info, use getTimeStamp instead).

perNyfelt committed:
Add (probably shaky) way of handling binary data. Add uniqueidentifier type to StringColumnBuilder
